### ISSUE TITLE
Error XA3006

### ISSUE THREAD

**Issue by kiavel** [xamarin/xamarin-android#8768]
### Android application type

Classic Xamarin.Android (MonoAndroid13.0, etc.)

### Affected platform version

VS 2022/VS MAC/Jetbrain Rider

### Description

Hello, 

I've attached an error that prevents me from compiling an Android application.
The .ll files are present in the obj folder.
They are located here: obj/Release(Debug)/monoandroid13.0/monoandroid13.0/android
and I think that the monoandroid13.0 repetition causes this problem, but I can't find a way to configure it.

### Steps to Reproduce

Don't know…

### Relevant log output

```shell
Xamarin.Android.Common.targets(2028,3) : Error XA3006 :  Could not compile native assembly file: environment.x86.ll

stderr | /Library/Frameworks/Xamarin.Android.framework/Libraries/xbuild/Xamarin/Android/Darwin/binutils/bin/llc : error : /Library/Frameworks/Xamarin.Android.framework/Libraries/xbuild/Xamarin/Android/Darwin/binutils/bin/llc : compressed_assemblies.arm64-v8a.ll : error : Could not open input file: no such file or directory


Xamarin.Android.Common.targets(2028,3) : Error XA3006 : Unable to compile native assembly file : marshal_methods.armeabi-v7a.ll

stderr | /Library/Frameworks/Xamarin.Android.framework/Libraries/xbuild/Xamarin/Android/Darwin/binutils/bin/llc : error : /Library/Frameworks/Xamarin.Android.framework/Libraries/xbuild/Xamarin/Android/Darwin/binutils/bin/llc : marshal_methods.armeabi-v7a.ll : error : Could not open input file: no such file or directory


Xamarin.Android.Common.targets(2028,3) : Error XA3006 : Unable to compile native assembly file : marshal_methods.arm64-v8a.ll

stderr | /Library/Frameworks/Xamarin.Android.framework/Libraries/xbuild/Xamarin/Android/Darwin/binutils/bin/llc : error : /Library/Frameworks/Xamarin.Android.framework/Libraries/xbuild/Xamarin/Android/Darwin/binutils/bin/llc : marshal_methods.arm64-v8a.ll : error : Could not open input file: no such file or directory
```

**Comment by jonathanpeppers** [comment]
Can you share a `.binlog` that contains the error? https://aka.ms/binlog

**Comment by kiavel** [comment]
Thank you for your reply, here is the file.

[ServiceRelease.binlog.zip](https://github.com/xamarin/xamarin-android/files/14422237/ServiceRelease.binlog.zip)

**Comment by jonathanpeppers** [comment]
Is XA3006 in this log? I'm not seeing it, as this is empty:
```powershell
> dotnet build -v:diag "D:\Downloads\ServiceRelease.binlog" | Select-String XA3006
```

One cause I noticed your projects are in `Z:\Proj_C#\` does the `#` in a directory name cause a problem? Is the log from Windows and the original report from a Mac?

**Comment by kiavel** [comment]
<img width="1842" alt="image" src="https://github.com/xamarin/xamarin-android/assets/7924179/f7483a2b-ee6f-4089-8510-b11afa655ce7">

I haven't Select-String command but without it, here is the result.
This binlog was build with Visual Studio Windows (ARM) because i don't know how to select msbuild 15 instead of 17 for build with macos terminal.
I tried to remove the # of my path (Z:\ProjC\CsXSuite\CsMiniWood\CsMiniWood.Android) and had the same error.
<img width="1976" alt="image" src="https://github.com/xamarin/xamarin-android/assets/7924179/dee5c4a0-7a06-4701-858e-38724d7a0b9c">

**Comment by dellis1972** [comment]
IS Z a networked drive? Historically tryingto build on networked drives has been very unstable. Can you try moving the entire project to a physical drive and see if that works.

**Comment by kiavel** [comment]
It's a networked drive because I'm working on parrallels desktop, but i've the same error on building directly on my MAC.
Same error when project is in C Drive.
<img width="1719" alt="image" src="https://github.com/xamarin/xamarin-android/assets/7924179/70689b26-01ca-4f8d-b152-d82ad2dec8ee">

**Comment by dellis1972** [comment]
This is going to sound like a silly question but you are deleting the `bin` and `obj` folders when switching between Mac and Windows right? We cache full paths and other stuff in the `obj` folder snd it will break if you try to build on windows with a mac cache.

**Comment by kiavel** [comment]
Of course, I deleted them. I've been searching for a solution to my problem for a while now. I've tried reverting to MonoAndroid 11 and 12, changing the SDK, and manually selecting various NDKs. I've looked into all possible properties of the csproj file to change the generation path of the .ll files. So, is it normal for there to be a double notation /MonoAndroid13/MonoAndroid13 in the project's obj directory?

**Comment by dellis1972** [comment]
> So, is it normal for there to be a double notation /MonoAndroid13/MonoAndroid13 in the project's obj directory?

Nope that is not standard.

**Comment by dellis1972** [comment]
As @jonathanpeppers mentioned we are going to need a full diagnostic .binlog to investigate the issue. The one provided is incomplete and does not contain the error you mentioned. It looks like the build was terminated while capturing the log as we get this error when trying to view it

```
There was an exception while reading the log file: BuildEvent record number 37179 (serialized size: 0) attempted to perform disallowed reads (details: System.IO.EndOfStreamException: Unable to read beyond the end of the stream.).
```

**Comment by kiavel** [comment]
is it okay with this ?

[ServiceRelease.zip](https://github.com/xamarin/xamarin-android/files/14435159/ServiceRelease.zip)

**Comment by kiavel** [comment]
<img width="757" alt="image" src="https://github.com/xamarin/xamarin-android/assets/7924179/7edbfac2-0d0a-4d8d-abc3-f8ad4d22d4b7">
It's strange, but the files do exist (and now they are in obj/release/monoandroid13.0/android).

**Comment by dellis1972** [comment]
Hmm, there is something odd going on with the `IntermediateOutputPath` value. 

```
Property reassignment: $(IntermediateOutputPath)="obj\Release\monoandroid13.0\" (previous value: "obj\Release\") at C:\Program Files\dotnet\sdk\8.0.200\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.DefaultOutputPaths.targets (142,5)
```

but in other areas of the log we see 

```
 _NativeAssemblySourceDir = obj\Release\android\
```

which is wrong. The property reassignment is messing up the build.

**Comment by grendello** [comment]
@dellis1972 it's normally generated by the `GeneratePackageManagerJava` task, and the log doesn't mention anything amiss with its invocation, so it's probably something wrong with the `obj/` paths, as you suspect.

**Comment by jonathanpeppers** [comment]
> Property reassignment: $(IntermediateOutputPath)="obj\Release\monoandroid13.0\" (previous value: "obj\Release\") at C:\Program Files\dotnet\sdk\8.0.200\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.DefaultOutputPaths.targets (142,5)

Wait, I'm confused. Why are we using the .NET 8 SDK for building a Xamarin.Android project? I think you'll have to use Mono msbuild, right?

**Comment by kiavel** [comment]
> > Property reassignment: $(IntermediateOutputPath)="obj\Release\monoandroid13.0" (previous value: "obj\Release") at C:\Program Files\dotnet\sdk\8.0.200\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.DefaultOutputPaths.targets (142,5)
> 
> Wait, I'm confused. Why are we using the .NET 8 SDK for building a Xamarin.Android project? I think you'll have to use Mono msbuild, right?

Does it mean that vs 2022 can not build Xamarin Android ?
I'm working on Jetbrains Rider (which can switch between MSBuild 15 and 17), and I have the same issue.
Here are the logs (I didn't yesterday how to build with binlog in rider).

[Archive.zip](https://github.com/xamarin/xamarin-android/files/14445652/Archive.zip)

**Comment by jonathanpeppers** [comment]
@kiavel to try Mono MSBuild (to see if it works):

```
msbuild YourAndroidProject.csproj -restore -t:SignAndroidPackage -bl
```
The only difference between `dotnet build` is you would need the `-restore` flag for NuGet packages. `-t:SignAndroidPackage` is also needed for Xamarin.Android application projects to build an `.apk`, but you don't need it for class libraries.

VS for Mac should be able to detect Xamarin projects and use Mono MSBuild, I thought?

**Comment by kiavel** [comment]
I did it, here is the binlog.

[ServiceRelease.binlog.zip](https://github.com/xamarin/xamarin-android/files/14450932/ServiceRelease.binlog.zip)

**Comment by jonathanpeppers** [comment]
The latest log looks more typical to me now. Does this problem occur for new Xamarin.Android projects (like the project template), or just this specific project?

**Comment by kiavel** [comment]
Thank for you help, I finally found the mistake. I compared the two .csproj files and found the difference...
My csproj file start with 
`<Project Sdk="Microsoft.NET.Sdk"> `
instead of 
`<Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">`

I don't know since when, but I think it's when I started the migration to MaUI.
Thank you all very much for your help.
